### PR TITLE
Planthunt

### PR DESCRIPTION
**Pull request by Zephyr75**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/api/CameraApi.java [hunk 1]
```diff
@@ -46,4 +46,8 @@ public static File createImageFile(Context context) throws IOException {
     public static String getImageUrl(FirebaseUser user, String fileName){
         return URL_FIRST + user.getUid() + "%2F" + fileName + URL_SECOND;
     }
+
+    public static String getNewImageUrl(FirebaseUser user, String fileName, String plantName){
+        return URL_FIRST + user.getUid() + "%2F" + plantName + "_" + fileName + URL_SECOND;
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/api/CameraApi.java:51]: What is this this %2F for ?

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java [hunk 2]
```diff
@@ -340,6 +346,121 @@ public void onSuccess(DataSnapshot dataSnapshot) {
...
 
+    /** Checks if lobby exists and given password matches correct one
+     *
+     * @param name      the unique identifier given to the lobby
+     */
+    public Task getLobbyPlayerCount (String name, OnCompleteListener<DataSnapshot> onCompleteListener){
+        return mDatabase
+                .child(LOBBIES)
+                .child(name)
+                .child(NBR_PLAYERS)
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java:357]: This function and the next one seem to have the same body with just a different String in the `.child` function call. Maybe you should create a common function ?

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Lobby.java [hunk 3]
```diff
@@ -38,30 +34,19 @@ public String getName(){
...
-            default:
-                status = FINISHED_STRING;
-                break;
-        }
-    }
 
     public int getNbrPlayers(){
         return nbrPlayers;
     }
+    public int getMaxNbrPlayers(){
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Lobby.java:41]: Shouldn't you add docs to those getters ?

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntJoinLobbyActivity.java [hunk 4]
```diff
@@ -37,22 +39,41 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                                    Log.e("ERROR", "Lobby does not exist!");
+                                }
+                                if (Math.toIntExact((long) task2.getResult().getValue()) < Math.toIntExact((long) task3.getResult().getValue())){
+                                    db.addUserToLobby(lobbyName, username);
 
+                                    //Launch lobby waiting screen
+                                    Intent intent = new Intent(PlanthuntJoinLobbyActivity.this, PlanthuntWaitLobbyActivity.class);
+                                    intent.putExtra(PlanthuntCreateJoinLobbyActivity.LOBBY_NAME, lobbyName);
+                                    intent.putExtra(PlanthuntCreateJoinLobbyActivity.USERNAME, username);
+                                    intent.putExtra(PlanthuntCreateJoinLobbyActivity.HOST, "player");
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntJoinLobbyActivity.java:66]: You should create a constant for this string.

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java [hunk 5]
```diff
@@ -1,16 +1,249 @@
...
+        }
+
+        db.mDatabase.child(Database.LOBBIES).child(lobbyName).child(Database.REMAINING_TIME).addValueEventListener(
+                new ValueEventListener() {
+                    @SuppressLint("SetTextI18n")
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot snapshot) {
+                        int time = Math.toIntExact((long) snapshot.getValue());
+                        String first = round(time / 60) < 10
+                                ? "0" + round(time / 60)
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java:103]: What does this code do for the `first` variable ?

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java [hunk 6]
```diff
@@ -1,16 +1,249 @@
...
+                }
+        );
+    }
+
+    private void startTimer(){
+        //Creates a new Thread to update timer asynchronously
+        Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                for (int i = 0; i < 301; i++) {
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java:150]: You should make a constant for the upperbound.

**Review by ozan-guven — CHANGES_REQUESTED** [review]
Great job ! You can look for my comments.

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java (hunk 5 above)
Comment (Zephyr75) [reply app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java:103]: It creates nice visuals for the timer.

File: app/src/main/java/ch/epfl/sdp/healthplay/api/CameraApi.java (hunk 1 above)
Comment (ozan-guven) [reply app/src/main/java/ch/epfl/sdp/healthplay/api/CameraApi.java:51]: I see now thanks.

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java (hunk 5 above)
Comment (ozan-guven) [reply app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java:103]: Oh good job !

**Review by ozan-guven — APPROVED** [review]
Good job ! Thanks for taking into consideration my comments.

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java [hunk 7]
```diff
@@ -1,7 +1,6 @@
 package ch.epfl.sdp.healthplay.database;
```
Comment (emonnin-epfl) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java:1]: Many of the comments for the lobby methods do not seem to match what is in the method.

**Review by emonnin-epfl — CHANGES_REQUESTED** [review]

**Review by emonnin-epfl — APPROVED** [review]

**Pull request merged**